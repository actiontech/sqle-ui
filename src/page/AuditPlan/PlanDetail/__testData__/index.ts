export const AuditPlanReportList = [
  {
    audit_plan_report_id: 31,
    audit_plan_report_timestamp: '2020-10-19T22:39:57+08:00',
  },
  {
    audit_plan_report_id: 32,
    audit_plan_report_timestamp: '2020-11-19T20:39:57+08:00',
  },
  {
    audit_plan_report_id: 33,
    audit_plan_report_timestamp: '2020-11-19T19:39:57+08:00',
  },
  {
    audit_plan_report_id: 34,
    audit_plan_report_timestamp: '2020-12-19T20:39:57+08:00',
  },
  {
    audit_plan_report_id: 35,
    audit_plan_report_timestamp: '2020-10-19T13:39:57+08:00',
  },
  {
    audit_plan_report_id: 36,
    audit_plan_report_timestamp: '2020-10-19T13:39:57+08:00',
  },
  {
    audit_plan_report_id: 37,
    audit_plan_report_timestamp: '2020-11-19T11:39:57+08:00',
  },
  {
    audit_plan_report_id: 38,
    audit_plan_report_timestamp: '2020-12-19T18:39:57+08:00',
  },
  {
    audit_plan_report_id: 39,
    audit_plan_report_timestamp: '2020-11-19T12:39:57+08:00',
  },
  {
    audit_plan_report_id: 40,
    audit_plan_report_timestamp: '2020-10-19T13:39:57+08:00',
  },
];

export const AuditPlanSqlsRes = {
  head: [
    { name: 'a', desc: '列a', type: 'sql' },
    { name: 'b', desc: '列b' },
  ],
  rows: [
    { a: 'select * from t1 where 1 = 1', b: 'b72' },
    { a: 'select * from t1 where 1 = 2', b: 'b74' },
    { a: 'select * from t1 where 1 = 3', b: 'b76' },
    { a: 'select * from t1 where 1 = 4', b: 'b78' },
    { a: 'select * from t1 where 1 = 5', b: 'b80' },
    { a: 'select * from t1 where 1 = 6', b: 'b82' },
    { a: 'select * from t1 where 1 = 7', b: 'b84' },
    { a: 'select * from t1 where 1 = 8', b: 'b86' },
    { a: 'select * from t1 where 1 = 9', b: 'b88' },
    { a: 'select * from t1 where 1 = 10', b: 'b90' },
  ],
};
export const AuditReport = [
  {
    audit_plan_report_sql: 'select * from t1 where id = t41',
    audit_plan_report_sql_audit_result:
      '[error]禁止使用没有where条件的sql语句或者使用where 1=1等变相没有条件的sql\n[notice]不建议使用select *\n[warn]CPU心情不太好\n[normal]所以CPU出门散心去了',
  },
  {
    audit_plan_report_sql: 'select * from t1 where id = t42',
    audit_plan_report_sql_audit_result:
      '[error]禁止使用没有where条件的sql语句或者使用where 1=1等变相没有条件的sql\n[notice]不建议使用select *\n[warn]CPU心情不太好\n[normal]所以CPU出门散心去了',
  },
  {
    audit_plan_report_sql: 'select * from t1 where id = t43',
    audit_plan_report_sql_audit_result:
      '[error]禁止使用没有where条件的sql语句或者使用where 1=1等变相没有条件的sql\n[notice]不建议使用select *\n[warn]CPU心情不太好\n[normal]所以CPU出门散心去了',
  },
  {
    audit_plan_report_sql: 'select * from t1 where id = t44',
    audit_plan_report_sql_audit_result:
      '[error]禁止使用没有where条件的sql语句或者使用where 1=1等变相没有条件的sql\n[notice]不建议使用select *\n[warn]CPU心情不太好\n[normal]所以CPU出门散心去了',
  },
  {
    audit_plan_report_sql: 'select * from t1 where id = t45',
    audit_plan_report_sql_audit_result:
      '[error]禁止使用没有where条件的sql语句或者使用where 1=1等变相没有条件的sql\n[notice]不建议使用select *\n[warn]CPU心情不太好\n[normal]所以CPU出门散心去了',
  },
  {
    audit_plan_report_sql: 'select * from t1 where id = t46',
    audit_plan_report_sql_audit_result:
      '[error]禁止使用没有where条件的sql语句或者使用where 1=1等变相没有条件的sql\n[notice]不建议使用select *\n[warn]CPU心情不太好\n[normal]所以CPU出门散心去了',
  },
  {
    audit_plan_report_sql: 'select * from t1 where id = t47',
    audit_plan_report_sql_audit_result:
      '[error]禁止使用没有where条件的sql语句或者使用where 1=1等变相没有条件的sql\n[notice]不建议使用select *\n[warn]CPU心情不太好\n[normal]所以CPU出门散心去了',
  },
  {
    audit_plan_report_sql: 'select * from t1 where id = t48',
    audit_plan_report_sql_audit_result:
      '[error]禁止使用没有where条件的sql语句或者使用where 1=1等变相没有条件的sql\n[notice]不建议使用select *\n[warn]CPU心情不太好\n[normal]所以CPU出门散心去了',
  },
  {
    audit_plan_report_sql: 'select * from t1 where id = t49',
    audit_plan_report_sql_audit_result:
      '[error]禁止使用没有where条件的sql语句或者使用where 1=1等变相没有条件的sql\n[notice]不建议使用select *\n[warn]CPU心情不太好\n[normal]所以CPU出门散心去了',
  },
  {
    audit_plan_report_sql: 'select * from t1 where id = t50',
    audit_plan_report_sql_audit_result:
      '[error]禁止使用没有where条件的sql语句或者使用where 1=1等变相没有条件的sql\n[notice]不建议使用select *\n[warn]CPU心情不太好\n[normal]所以CPU出门散心去了',
  },
];
