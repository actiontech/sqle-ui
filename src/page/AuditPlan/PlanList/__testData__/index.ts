export const AuditPlanList = [
  {
    audit_plan_cron: '0 */2 * * *',
    audit_plan_db_type: 'mysql',
    audit_plan_instance_database: 'sqle',
    audit_plan_instance_name: 'db1',
    audit_plan_name: 'audit_for_java_app11',
    audit_plan_token:
      'wkpjrnbuneufaljqxijggisjedhpwlhippscklduloxbgymgvwkvhuurmyjiqbaiufvadmkncenueesmbqjqeudkbqdhvihiwzrsdrqbluhashusgzemonxivbsrz',
    audit_plan_meta: {
      audit_plan_params: [],
      audit_plan_type: 'audit_for_java_app',
      audit_plan_type_desc: '审核java应用',
      instance_type: 'mysql',
    },
  },
  {
    audit_plan_cron: '0 */2 * * *',
    audit_plan_db_type: 'oracle',
    audit_plan_instance_database: 'sqle',
    audit_plan_instance_name: 'db1',
    audit_plan_name: 'audit_for_java_app12',
    audit_plan_token:
      'itfpkfctmologkcspmlxwugumdvfslgbgcisqczpehskfjprctfkeuebywxrwsnseltrmsnpwrilahilbxeycpksttfnfbevkrsundytiflrvfccthmrnmorjqfso',
    audit_plan_meta: {
      audit_plan_params: [],
      audit_plan_type: 'audit_for_java_app',
      audit_plan_type_desc: '审核java应用',
      instance_type: 'oracle',
    },
  },
  {
    audit_plan_cron: '0 */2 * * *',
    audit_plan_db_type: 'mysql',
    audit_plan_instance_database: 'sqle',
    audit_plan_instance_name: 'db1',
    audit_plan_name: 'audit_for_java_app13',
    audit_plan_token:
      'aqulzprcpvtydznpplswiijhxsufviotqtfxkocpoukqsiugkmpsnmslnftfdeynbiiobvsguanlktvupflxqaeyjsftedqsuwmxabixdhpekbehnxxqrrjolncmw',
    audit_plan_meta: {
      audit_plan_params: [],
      audit_plan_type: 'audit_for_node_app',
      audit_plan_type_desc: '审核node应用',
      instance_type: 'oracle',
    },
  },
  {
    audit_plan_cron: '0 */2 * * *',
    audit_plan_db_type: 'oracle',
    audit_plan_instance_database: 'sqle',
    audit_plan_instance_name: 'db1',
    audit_plan_name: 'audit_for_java_app14',
    audit_plan_token:
      'nsucohfvciisgsisgythudugjianedqdwvvlefrclxydqxryffrwsunwlhclrboagskhhqpgqthkrwbqhwgziqrdxxswoltlocxaholiwoqpouknqcwelxbekdmjl',
    audit_plan_meta: {
      audit_plan_params: [],
      audit_plan_type: 'slow_log_audit',
      audit_plan_type_desc: '审核慢日志',
      instance_type: 'oracle',
    },
  },
  {
    audit_plan_cron: '0 */2 * * *',
    audit_plan_db_type: 'mysql',
    audit_plan_instance_database: 'sqle',
    audit_plan_instance_name: 'db1',
    audit_plan_name: 'audit_for_java_app15',
    audit_plan_token:
      'oobxrdhaggcwrncxctgtrkmpsrhqlscrxqrrwjqdinxvehvcnhfjmnbefrdmfsuwwfwncqgynnjmwhperpwmmnyhnrszrusrtyaywfuzklnbfvtiaeuxqqhfssuiu',
    audit_plan_meta: {
      audit_plan_params: [],
      audit_plan_type: 'mybatis_audit',
      audit_plan_type_desc: '审核mybatis应用',
      instance_type: 'oracle',
    },
  },
  {
    audit_plan_cron: '0 */2 * * *',
    audit_plan_db_type: 'oracle',
    audit_plan_instance_database: 'sqle',
    audit_plan_instance_name: 'db1',
    audit_plan_name: 'audit_for_java_app16',
    audit_plan_token:
      'ecylwymxtxpvqhkqmtlgqubfudpmurqofvsrvrviihnmzwgtukvqsljvwqdbpxoijeqzvjjttkcmisfspscldihwbktuzuikwxbcstlntldconobxgnqnvoizkunp',
  },
  {
    audit_plan_cron: '0 */2 * * *',
    audit_plan_db_type: 'mysql',
    audit_plan_instance_database: 'sqle',
    audit_plan_instance_name: 'db1',
    audit_plan_name: 'audit_for_java_app17',
    audit_plan_token:
      'rxgloynyjuvfpeghqgkuxcolghxkwmiiebxweiqryuftjpepbshmbqitcloeijaxunwqghlrqytobdrtvkrinsmmpvdvjrhvqnogbbkkrvgxvrmfcjhgsbhjjrvpj',
  },
  {
    audit_plan_cron: '0 */2 * * *',
    audit_plan_db_type: 'oracle',
    audit_plan_instance_database: 'sqle',
    audit_plan_instance_name: 'db1',
    audit_plan_name: 'audit_for_java_app18',
    audit_plan_token:
      'psudtpmsqebdnxjxdtmtripsexqcpvrdhydtnldhslxagcvftnyixhwhurunpqylnrbtgfyxssbjkvfysmojwrvxrqqztutnykjrchjhqasolftsweirknvjrrvlt',
  },
  {
    audit_plan_cron: '0 */2 * * *',
    audit_plan_db_type: 'mysql',
    audit_plan_instance_database: 'sqle',
    audit_plan_instance_name: 'db1',
    audit_plan_name: 'audit_for_java_app19',
    audit_plan_token:
      'tmlgcxbmpcniasdmkqarjereoonwqtohdnfxijmmsuqqkdecclqtbrfrfdnlkxtpwbqlxbzwjwcjfhukjonrjdevqowzcmsvxhgigprwfrkqahwidmjusouiarurg',
  },
  {
    audit_plan_cron: '0 */2 * * *',
    audit_plan_db_type: 'oracle',
    audit_plan_instance_database: 'sqle',
    audit_plan_instance_name: 'db1',
    audit_plan_name: 'audit_for_java_app20',
    audit_plan_token:
      'tidubrvskubhlwtqqexgnjqehlvrgoggmcmwokwetcpmocxyyohgcrctijsmkgiwisetjnursctznwcqayztiwnjnihmphfegfnoubexphwhzzgzjhufculijanuv',
  },
];

export const AuditPlan = {
  audit_plan_cron: '* * * * *',
  audit_plan_db_type: 'oracle',
  audit_plan_instance_database: 'sqle',
  audit_plan_instance_name: 'db1',
  audit_plan_name: 'audit_for_java_app20',
  audit_plan_token:
    'tidubrvskubhlwtqqexgnjqehlvrgoggmcmwokwetcpmocxyyohgcrctijsmkgiwisetjnursctznwcqayztiwnjnihmphfegfnoubexphwhzzgzjhufculijanuv',
  audit_plan_meta: {
    audit_plan_params: [
      { desc: '字段a', key: 'a', type: 'string', value: '123' },
      { desc: '字段b', key: 'b', type: 'int', value: '123' },
      { desc: '字段c', key: 'c', type: 'bool', value: 'true' },
    ],
    audit_plan_type: 'normal',
    audit_plan_type_desc: '普通的SQL审核',
    instance_type: 'oracle',
  },
};
