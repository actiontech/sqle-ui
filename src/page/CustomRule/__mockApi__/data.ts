import { ICustomRuleResV1 } from '../../../api/common';
import { CustomRuleResV1LevelEnum } from '../../../api/common.enum';

export const customRules: ICustomRuleResV1[] = [
  {
    db_type: 'mysql',
    annotation: 'desc desc desc',
    level: CustomRuleResV1LevelEnum.notice,
    rule_id: '1',
    desc: 'name1',
    rule_script: 'w+',
    type: 'rule_type1',
  },
  {
    db_type: 'oracle',
    annotation: 'desc desc desc',
    level: CustomRuleResV1LevelEnum.error,
    rule_id: '2',
    desc: 'name2',
    rule_script: 'w+',
    type: 'rule_type2',
  },
];
