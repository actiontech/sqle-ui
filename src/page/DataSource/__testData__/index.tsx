export const dataSourceList = [
  {
    instance_name: 'db1',
    db_host: '20.20.20.2',
    db_port: '3306',
    db_user: 'root',
    desc: '',
    workflow_template_name: 'default',
  },
];

export const dataSourceInstance = {
  instance_name: 'db1',
  db_host: '20.20.20.2',
  db_port: '3306',
  db_user: 'root',
  db_type: 'mysql',
  desc: '',
  workflow_template_name: 'workflow-template-name-1',
};

export const dataSourceMetas = [
  {
    db_type: 'mysql',
    params: [
      { description: '字段a', name: 'a', type: 'string', value: '123' },
      { description: '字段b', name: 'b', type: 'int', value: '123' },
      { description: '字段c', name: 'c', type: 'bool', value: 'true' },
    ],
  },
  {
    db_type: 'oracle',
    params: [
      { description: '字段e', name: 'e', type: 'string', value: '2222' },
      { description: '字段f', name: 'f', type: 'int', value: '333' },
      { description: '字段mm', name: 'mm', type: 'bool', value: 'false' },
    ],
  },
];
