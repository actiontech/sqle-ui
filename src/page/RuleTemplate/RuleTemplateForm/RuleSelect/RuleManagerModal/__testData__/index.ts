import { IRuleResV1 } from '../../../../../../api/common';
import {
  RuleParamResV1TypeEnum,
  RuleResV1LevelEnum,
} from '../../../../../../api/common.enum';
export const ruleData: IRuleResV1 = {
  rule_name: 'ddl_check_collation_database',
  desc: '建议使用规定的数据库排序规则',
  level: RuleResV1LevelEnum.error,
  type: 'DDL规范',
  db_type: 'mysql',
  annotation: 'annotation',
};

export const editRuleData: IRuleResV1 = {
  rule_name: 'ddl_check_collation_database',
  desc: '建议使用规定的数据库排序规则',
  params: [
    {
      value: 'str',
      type: RuleParamResV1TypeEnum.string,
      key: 'str_key',
      desc: 'str_desc',
    },
    {
      value: '123',
      type: RuleParamResV1TypeEnum.int,
      key: 'int_key',
      desc: 'ine_desc',
    },
    {
      value: 'false',
      type: RuleParamResV1TypeEnum.bool,
      key: 'radio_key',
      desc: 'radio_desc',
    },
  ],
  level: RuleResV1LevelEnum.normal,
  type: 'DDL规范',
  db_type: 'mysql',
};
