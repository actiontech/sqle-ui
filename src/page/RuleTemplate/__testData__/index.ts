export const ruleTemplateListData = [
  {
    rule_template_name: 'default',
    desc: '默认规则模板',
    rule_name_list: [
      'all_check_where_is_invalid',
      'ddl_check_alter_table_need_merge',
      'ddl_check_column_blob_default_is_not_null',
      'ddl_check_column_blob_with_not_null',
      'ddl_check_column_char_length',
      'ddl_check_column_timestamp_without_default',
      'ddl_check_column_varchar_max',
      'ddl_check_column_without_comment',
      'ddl_check_column_without_default',
      'ddl_check_composite_index_max',
      'ddl_check_index_column_with_blob',
      'ddl_check_index_count',
      'ddl_check_index_prefix',
      'ddl_check_object_name_length',
      'ddl_check_object_name_using_cn',
      'ddl_check_object_name_using_keyword',
      'ddl_check_pk_not_exist',
      'ddl_check_pk_without_auto_increment',
      'ddl_check_pk_without_bigint_unsigned',
      'ddl_check_table_without_comment',
      'ddl_check_table_without_if_not_exists',
      'ddl_check_table_without_innodb_utf8mb4',
      'ddl_check_unique_index_prefix',
      'ddl_disable_drop_statement',
      'ddl_disable_fk',
      'ddl_osc_min_size',
      'dml_check_with_limit',
      'dml_check_with_order_by',
      'dml_disable_select_all_column',
      'dml_rollback_max_rows',
    ],
  },
  {
    rule_template_name: 'default2',
    desc: '默认规则模板2',
    rule_name_list: [
      'ddl_osc_min_size',
      'dml_check_with_limit',
      'dml_check_with_order_by',
      'dml_disable_select_all_column',
      'dml_rollback_max_rows',
    ],
  },
];

export const ruleTemplateData = {
  rule_template_name: 'default',
  desc: '默认规则模板',
  rule_name_list: [
    'all_check_where_is_invalid',
    'ddl_check_alter_table_need_merge',
    'ddl_check_column_blob_default_is_not_null',
    'ddl_check_column_blob_with_not_null',
    'ddl_check_column_char_length',
    'ddl_check_column_timestamp_without_default',
    'ddl_check_column_varchar_max',
    'ddl_check_column_without_comment',
    'ddl_check_column_without_default',
    'ddl_check_composite_index_max',
    'ddl_check_index_column_with_blob',
    'ddl_check_index_count',
    'ddl_check_index_prefix',
    'ddl_check_object_name_length',
    'ddl_check_object_name_using_cn',
    'ddl_check_object_name_using_keyword',
    'ddl_check_pk_not_exist',
    'ddl_check_pk_without_auto_increment',
    'ddl_check_pk_without_bigint_unsigned',
    'ddl_check_table_without_comment',
    'ddl_check_table_without_if_not_exists',
    'ddl_check_table_without_innodb_utf8mb4',
    'ddl_check_unique_index_prefix',
    'ddl_disable_drop_statement',
    'ddl_disable_fk',
    'ddl_osc_min_size',
    'dml_check_with_limit',
    'dml_check_with_order_by',
    'dml_disable_select_all_column',
    'dml_rollback_max_rows',
  ],
};
