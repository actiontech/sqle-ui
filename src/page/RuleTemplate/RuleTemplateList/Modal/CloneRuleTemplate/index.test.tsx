import { fireEvent, screen, cleanup, waitFor } from '@testing-library/react';
import { useParams } from 'react-router-dom';
import CloneRuleTemplateModal from '.';
import rule_template from '../../../../../api/rule_template';
import EmitterKey from '../../../../../data/EmitterKey';
import { ModalName } from '../../../../../data/ModalName';
import { renderWithRouter } from '../../../../../testUtils/customRender';
import {
  mockUseDispatch,
  mockUseSelector,
} from '../../../../../testUtils/mockRedux';
import {
  mockUseInstance,
  resolveThreeSecond,
} from '../../../../../testUtils/mockRequest';
import EventEmitter from '../../../../../utils/EventEmitter';
import { ruleTemplateListData } from '../../../__testData__';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: jest.fn(),
}));
const projectName = 'default';

describe('RuleTemplate/RuleTemplateList/Modal/CloneRuleTemplateModal', () => {
  let mockDispatch: jest.Mock;
  const useParamsMock: jest.Mock = useParams as jest.Mock;

  beforeEach(() => {
    jest.useFakeTimers();
    const { scopeDispatch } = mockUseDispatch();
    mockUseInstance();
    mockUseSelector({
      ruleTemplate: {
        modalStatus: { [ModalName.Clone_Rule_Template]: true },
        selectRuleTemplate: ruleTemplateListData[0],
      },
    });
    mockDispatch = scopeDispatch;
    useParamsMock.mockReturnValue({ projectName });
  });

  afterEach(() => {
    jest.useRealTimers();
    jest.clearAllMocks();
    jest.clearAllTimers();
  });

  const mockCloneRuleTemplate = () => {
    const spy = jest.spyOn(rule_template, 'cloneProjectRuleTemplateV1');
    spy.mockImplementation(() => resolveThreeSecond({}));
    return spy;
  };

  test('should match snapshot', async () => {
    const { baseElement: nullElement } = renderWithRouter(
      <CloneRuleTemplateModal />
    );
    expect(nullElement).toMatchSnapshot();
    cleanup();
    mockUseSelector({
      ruleTemplate: {
        modalStatus: { [ModalName.Clone_Rule_Template]: false },
        selectRuleTemplate: ruleTemplateListData[0],
      },
    });
    const { baseElement } = renderWithRouter(<CloneRuleTemplateModal />);
    expect(baseElement).toMatchSnapshot();
  });

  test('should close modal when use click cancel button', async () => {
    renderWithRouter(<CloneRuleTemplateModal />);
    fireEvent.input(
      screen.getByLabelText('ruleTemplate.ruleTemplateForm.templateDesc'),
      { target: { value: 'desc1' } }
    );
    fireEvent.click(screen.getByText('Cancel'));
    expect(
      screen.getByLabelText('ruleTemplate.ruleTemplateForm.templateDesc')
    ).toHaveValue('');
    expect(mockDispatch).toBeCalledTimes(1);
    expect(mockDispatch).toBeCalledWith({
      payload: {
        modalName: 'CLONE_RULE_TEMPLATE',
        status: false,
      },
      type: 'ruleTemplate/updateModalStatus',
    });
  });

  test('should send clone template request when user click submit button', async () => {
    const cloneRequestSpy = mockCloneRuleTemplate();
    renderWithRouter(<CloneRuleTemplateModal />);
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });

    fireEvent.input(
      screen.getByLabelText('ruleTemplate.ruleTemplateForm.templateName'),
      { target: { value: 'name1' } }
    );
    fireEvent.input(
      screen.getByLabelText('ruleTemplate.ruleTemplateForm.templateDesc'),
      { target: { value: 'desc1' } }
    );

    fireEvent.mouseDown(
      screen.getByLabelText('ruleTemplate.ruleTemplateForm.instances')
    );
    await waitFor(() => {
      jest.advanceTimersByTime(0);
    });
    const option = screen.getAllByText('instance1')[1];
    fireEvent.click(option);

    fireEvent.click(screen.getByText('OK'));
    await waitFor(() => {
      jest.advanceTimersByTime(0);
    });
    expect(cloneRequestSpy).toBeCalledTimes(1);
    expect(cloneRequestSpy).toBeCalledWith({
      desc: 'desc1',
      rule_template_name: ruleTemplateListData[0].rule_template_name,
      new_rule_template_name: 'name1',
      instance_name_list: ['instance1'],
      project_name: projectName,
    });
    expect(screen.getByText('OK').parentNode).toHaveClass('ant-btn-loading');
    expect(screen.getByText('Cancel').parentNode).toHaveAttribute('disabled');

    const emitSpy = jest.spyOn(EventEmitter, 'emit');
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });

    expect(emitSpy).toBeCalledTimes(1);
    expect(emitSpy).toBeCalledWith(EmitterKey.Refresh_Rule_Template_List);
    expect(
      screen.queryByText('ruleTemplate.cloneRuleTemplate.successTips')
    ).toBeInTheDocument();
    expect(
      screen.getByLabelText('ruleTemplate.ruleTemplateForm.templateName')
    ).toHaveValue('');
    expect(mockDispatch).toBeCalledTimes(1);
    expect(mockDispatch).toBeCalledWith({
      payload: {
        modalName: 'CLONE_RULE_TEMPLATE',
        status: false,
      },
      type: 'ruleTemplate/updateModalStatus',
    });
  });
});
