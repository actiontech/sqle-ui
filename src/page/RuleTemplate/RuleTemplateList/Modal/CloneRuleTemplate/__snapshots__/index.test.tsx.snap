// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RuleTemplate/RuleTemplateList/Modal/CloneRuleTemplateModal should match snapshot 1`] = `
<body
  class="ant-scrolling-effect"
  style="width: calc(100% - 0px); overflow: hidden; overflow-x: hidden; overflow-y: hidden;"
>
  <div />
  <div>
    <div
      class="ant-modal-root"
    >
      <div
        class="ant-modal-mask"
      />
      <div
        aria-labelledby="rcDialogTitle0"
        class="ant-modal-wrap"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="ant-modal"
          role="document"
          style="width: 520px;"
        >
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
          <div
            class="ant-modal-content"
          >
            <div
              class="ant-modal-header"
            >
              <div
                class="ant-modal-title"
                id="rcDialogTitle0"
              >
                ruleTemplate.cloneRuleTemplate.title
              </div>
            </div>
            <div
              class="ant-modal-body"
            >
              <div
                class="ant-space ant-space-vertical full-width-element"
              >
                <div
                  class="ant-space-item"
                  style="margin-bottom: 8px;"
                >
                  <div
                    class="ant-row"
                  >
                    <div
                      class="ant-col ant-col-offset-2"
                    >
                      ruleTemplate.cloneRuleTemplate.currentTemplateTips
                      <a
                        href="/rule/template/update/default_mysql"
                        target="_blank"
                      >
                        default_mysql
                      </a>
                    </div>
                    <div
                      class="ant-col ant-col-offset-2"
                    >
                      <span
                        class="ant-typography ant-typography-secondary"
                      >
                        ruleTemplate.cloneRuleTemplate.cloneDesc
                      </span>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-space-item"
                >
                  <form
                    class="ant-form ant-form-horizontal"
                  >
                    <div
                      class="ant-row ant-form-item"
                    >
                      <div
                        class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-7"
                      >
                        <label
                          class="ant-form-item-required"
                          for="templateName"
                          title="ruleTemplate.ruleTemplateForm.templateName"
                        >
                          ruleTemplate.ruleTemplateForm.templateName
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-13"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <input
                              class="ant-input"
                              id="templateName"
                              placeholder="common.form.placeholder.input"
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="ant-row ant-form-item"
                    >
                      <div
                        class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-7"
                      >
                        <label
                          class=""
                          for="templateDesc"
                          title="ruleTemplate.ruleTemplateForm.templateDesc"
                        >
                          ruleTemplate.ruleTemplateForm.templateDesc
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-13"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <textarea
                              class="ant-input textarea-no-resize"
                              id="templateDesc"
                              placeholder="common.form.placeholder.input"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="ant-row ant-form-item"
                    >
                      <div
                        class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-7"
                      >
                        <label
                          class=""
                          for="instances"
                          title="ruleTemplate.ruleTemplateForm.instances"
                        >
                          ruleTemplate.ruleTemplateForm.instances
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-13"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <div
                              class="ant-select ant-select-multiple ant-select-allow-clear ant-select-show-search"
                            >
                              <div
                                class="ant-select-selector"
                              >
                                <div
                                  class="ant-select-selection-overflow"
                                >
                                  <div
                                    class="ant-select-selection-overflow-item ant-select-selection-overflow-item-suffix"
                                    style="opacity: 1;"
                                  >
                                    <div
                                      class="ant-select-selection-search"
                                      style="width: 0px;"
                                    >
                                      <input
                                        aria-activedescendant="instances_list_0"
                                        aria-autocomplete="list"
                                        aria-controls="instances_list"
                                        aria-haspopup="listbox"
                                        aria-owns="instances_list"
                                        autocomplete="off"
                                        class="ant-select-selection-search-input"
                                        id="instances"
                                        readonly=""
                                        role="combobox"
                                        style="opacity: 0;"
                                        type="search"
                                        unselectable="on"
                                        value=""
                                      />
                                      <span
                                        aria-hidden="true"
                                        class="ant-select-selection-search-mirror"
                                      >
                                        
                                        Â 
                                      </span>
                                    </div>
                                  </div>
                                </div>
                                <span
                                  class="ant-select-selection-placeholder"
                                >
                                  common.form.placeholder.select
                                </span>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </form>
                </div>
              </div>
            </div>
            <div
              class="ant-modal-footer"
            >
              <button
                class="ant-btn"
                type="button"
              >
                <span>
                  Cancel
                </span>
              </button>
              <button
                class="ant-btn ant-btn-primary"
                type="button"
              >
                <span>
                  OK
                </span>
              </button>
            </div>
          </div>
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`RuleTemplate/RuleTemplateList/Modal/CloneRuleTemplateModal should match snapshot 2`] = `
<body
  class=""
  style=""
>
  <div />
</body>
`;
